movies-manager/src/containers/main/main.tsx

```diff
@@ -161,35 +148,65 @@ export function Main(props: IMainProps): JSX.Element {
             </li>;
         });
 
-    return <main className={blockName}>
+    return loadConfig.isLoading ? <LoadingIndicator /> :
+        !loadConfig.isLoaded && <ErrorHandler errorMessage={errorInfo.message}/>
+        || <main className={blockName}>
         <Modal isOpen={isFormDialogOpen} handleClose={hideModal}>
-            <FormPage onSaveChanges={updateMoviesSet} movie={ movieToEdit }/>
+            <FormPageWithState onSaveChanges={updateMoviesSet} movie={ movieToEdit }/>
         </Modal>
         <Modal isOpen={isDeleteDialogOpen} handleClose={hideModal}>
-            <DeleteModal  onDeleteConfirm={deleteMovie} title={movieToEdit.title}/>
+            <DeleteModal onDeleteConfirm={onDeleteMovie} title={movieToEdit.title}/>
         </Modal>
         <Wrapper>
-            { props.areDetailsVisible && movieWithDetails ?
-                <Details { ...movieWithDetails }/> : <Search/> }
+            { props.areDetailsVisible ? <DetailsWithState /> : <Search onSearchClick={searchMovies} /> }
         </Wrapper>
         <div className={`${blockName}__separator`} />
         <Wrapper>
             <>
                 <section className={`${blockName}__filter`}>
                     <ResultFilter
-                        onGenreClick={setCurrentGenre}
-                        { ...moviesGenresConfig }/>
+                        onGenreClick={sortMoviesByGenre}
+                        { ...storeGenresConfig }/>
                     <ResultSort
-                        onSortClick={updateMoviesSortConfig}
+                        onSortClick={sortMoviesByField}
+                        showOptionList={isSortByOpen}
                         {...moviesSortConfig}/>
                 </section>
                 <div className={`${blockName}__amount`}>
-                    <strong className='strong'>{movies.length}</strong> movies found
+                    <strong className='strong'>{moviesStore.length}</strong> movies found
                 </div>
                 <ul className={`${blockName}__cards-list`}>
                     {moviesCards}
                 </ul>
             </>
         </Wrapper>
     </main>;
-}
\ No newline at end of file
+}
+
+const mapStateToProps = (state: IStoreState, ownProps: IMainProps) => {
+    return {
+      ...ownProps,
+      moviesStore: state.moviesConfig.movies,
+      loadConfig: {
+          isLoaded: state.moviesConfig.isLoaded,
+          isLoading: state.moviesConfig.isLoading,
+      },
+      errorInfo: state.moviesConfig.error,
+      storeGenresConfig: state.filters.genresConfig,
+      moviesSortConfig: state.filters.sortByConfig,
+    };
+};
+
+const dispatchToProps = ((dispatch: any) => {
+    return {
+        loadData: (params: TNullable<IQueryParams>) => { dispatch(getMoviesFromServer(params)); },
+        deleteMovie: (id: number) => { dispatch(deleteMoviesFromServer(id)); },
+        setCurrentGenre: (genre: TGenresListItem) => { dispatch(setGenreFilter(genre)); },
```

Comment:
setCurrentGenre should be in ResultFilter

Reply:
Almost the same reason, as previous. I do not want to do smth with store in dumb components. By the way I fetch data, when click sort in such a way it will be neсessary to place logic for the same action in 2 different components.

Reply:
Taking into account 2 upper your discusstions the only thing I can suppose - separate additional component FilterSet and place tabs with genres, sortBy and search in it.